build/js/main.js

```diff
@@ -1177,89 +1199,80 @@ class Bluetooth extends Plugin {
 
     scanForDevices() {
         this.status(`Start scanning`);
-        api.putPlugin(this.callsign, 'Scan', null, (err, resp) => {
-            if (err !== null) {
-                console.error(err);
-                return;
-            }
-
-            // update after 2s
-            setTimeout(this.update.bind(this), 2000);
-
-            // update discovered device list in every 4s
-            this.Timer = setInterval(this.getDiscoveredDevices.bind(this), 4000);
-
-            this.status(`Scanning...`);
-            // stop scan after 1 min
-            setTimeout(this.stopScan.bind(this), 60000);
-
-        });
-    }
-
-    stopScan() {
-        this.status(`Stopping Scan`);
-        clearInterval(this.Timer);
-        api.putPlugin(this.callsign, 'StopScan', null, (err, resp) => {
+        var f = document.getElementById("BT_DeviceType");
+        var device = f.options[f.selectedIndex].value;
+        api.putPlugin(this.callsign, 'Scan/?LowEnergy='+device, null, (err, resp) => {
             if (err !== null) {
                 console.error(err);
                 return;
             }
-
-            setTimeout(this.update.bind(this), 2000);
-            this.status(`Scan stopped`);
+	    document.getElementById('BT_ScanForDevices').disabled = true;
+            document.getElementById('BT_Pair').disabled = false;
+            setTimeout(this.update.bind(this), 100);
+            // update every 3s
+            this.Timer = setInterval(this.update.bind(this), 3000);
         });
     }
 
     pairDevice() {
-        var val = JSON.parse(document.getElementById('BT_DiscoveredDevices').value);
-        if (val.Name === '')
-            this.status(`Pairing with ${val.Address}`);
+        var idx = this.discoveredDeviceList.selectedIndex;
+        if (this.discoveredDevices[idx].name === "")
+            this.status(`Pairing to ${this.discoveredDevices[idx].address}`);
         else
-            this.status(`Pairing with ${val.Name}`);
+            this.status(`Pairing to ${this.discoveredDevices[idx].name}`);
 
-        api.putPlugin(this.callsign, 'Pair', '{"Address" : "' + val.Address + '"}', (err, resp) => {
-            if (err !== null) {
+        api.putPlugin(this.callsign, 'Pair', '{"address" : "' + this.discoveredDevices[idx].address + '"}', (err,resp) =>{
+
```

Comment:
Is there already a JSONRPC interface available for bluetooth? Else consider using `api.req` method to provide backwards compatibility during the transition phase.